homeassistant/components/lg_thinq/number.py

```diff
@@ -117,7 +117,16 @@
     DeviceType.WASHTOWER_DRYER: WASHER_NUMBERS,
     DeviceType.WASHTOWER: WASHER_NUMBERS,
     DeviceType.WASHTOWER_WASHER: WASHER_NUMBERS,
-    DeviceType.WATER_HEATER: (NUMBER_DESC[ThinQProperty.TARGET_TEMPERATURE],),
+    DeviceType.WATER_HEATER: (
+        NumberEntityDescription(
+            key=ThinQProperty.TARGET_TEMPERATURE,
+            native_max_value=60,
+            native_min_value=35,
+            native_step=1,
+            native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+            translation_key=ThinQProperty.TARGET_TEMPERATURE,
+        ),
+    ),
```

Comment:
Why didn't we implement the water_heater platform for this?

Reply:
HA water_heater platform support only 0.5 step.
But our devices only support step 1.
That's why we didn't implement the water_header platform.

Reply:
You can overwrite the precision right?

https://github.com/home-assistant/home-assistant/blob/master/homeassistant%2Fcomponents%2Fwater_heater%2F__init__.py#L194-L201

Reply:
Let us discuss about that.

Reply:
As we tested in HomeAssistant, frontend only supporty 0.5 step with +,- buttons.
That is because water_heater entity does not have 'attr_target_temp_step'.
We need 0.5 step in frontend.
If I am wrong please let me know.

Reply:
Frontend has support but backend not (`target_temp_step` is not implemented in `WaterHeaterEntity`).
Otherwise for Fahrenheit it will be step 1, for celsius 0.5.

Reply:
In conclusion, WaterHeaterEntity does not work as we want.
So we need this.

Reply:
I think we can agree with that so I close this conversation as resolved.